packages/wells/src/__tests__/integration/wells.int.spec.ts

```diff
@@ -31,21 +41,46 @@ describeIfCondition('CogniteClient setup in wells - integration test', () => {
     expect(response.length).toBe(2);
   });
 
-  test('get empty wells array with name suffix', async () => {
+  test('custom filter - get well by asset name prefix', async () => {
+    const namePrefix = 'Well';
+
+    const fn: SearchWell = async (args: string) =>
+      await client.wells.getWellsByNamePrefix(args);
+
+    const response = await client.wells.getWellsByNamePrefix(namePrefix, fn);
+    response.forEach(function(well) {
+      expect(well.name.startsWith('Well')).toBe(true);
+    });
+    expect(response.length).toBe(2);
+  });
```

Comment:
@rek example of how you can override with  a custom function `fn: SearchWell = (args) => someCustomMethod(args)`